### ISSUE TITLE
BUG: numpy import error in 'electromagnetics/natural_source/utils/plot_data_types.py'

### ISSUE THREAD

**Issue by pjdsmedt** [simpeg/simpeg#1505]
### Describe the issue:

when importing frequencydomain from simpeg.electromagnetics, an error is thrown due to erroneous importing of numpy:

`from SimPEG.electromagnetics import frequency_domain` gives: 

> simpeg/electromagnetics/natural_source/utils/plot_data_types.py:1
----> site-packages/simpeg/electromagnetics/natural_source/utils/plot_data_types.py: from matplotlib import pyplot as plt, colors, numpy as np
  (...)
ImportError: cannot import name 'numpy' from 'matplotlib'

in 'simpeg/electromagnetics/natural_source/utils/plot_data_types.py', correct the import line `from matplotlib import pyplot as plt, colors, numpy as np` -> `from matplotlib import pyplot as plt, colors`  and `import numpy as np`

@santisoler : I see this has likely been addressed in [Improve imports in natural source utils #1503](https://github.com/simpeg/simpeg/pull/1503), but the error occurs on simpeg 0.22, released last week. I assume the updates just need to be implemented?

### Reproducible code example:

```python
from SimPEG.electromagnetics import frequency_domain
```

### Error message:

```shell
ImportError: cannot import name 'numpy' from 'matplotlib'
```

### Runtime information:

OS | Darwin (macOS 14.5) | CPU(s) | 8 | Machine | x86_64
-- | -- | -- | -- | -- | --
Architecture | 64bit | RAM | 16.0 GiB | Environment | Jupyter
File system | apfs
Python 3.12.4 \| packaged by conda-forge \| (main, Jun 17 2024, 10:11:10) [Clang 16.0.6 ]
simpeg | 0.22.0 | discretize | 0.10.0 | pymatsolver | 0.2.0
numpy | 1.26.4 | scipy | 1.13.1 | sklearn | 1.4.2
matplotlib | 3.9.0 | empymod | 2.3.1 | geoana | 0.6.0
pandas | 2.2.2 | numba | 0.59.1 | IPython | 8.26.0

**Comment by santisoler** [comment]
Thanks @pjdsmedt for reporting this. What's going on is that the `plot_data_types.py` file relies on `matplotlib` importing `numpy` in their base `__init__.py`. This wasn't a problem with previous versions of `matplotlib`, but since 3.9.0 it's not importing `numpy` in `matplotlib/__init__.py`. That's why you're experiencing the error.

I'll try to release a simpeg patch (v0.22.1) soon. A temporary solution I can offer you is to pin `matplotlib<3.9.0`, at least until we release the patch.

**Comment by pjdsmedt** [comment]
No problem. Thanks for the info!

**Comment by santisoler** [comment]
This bug has been fixed in #1503 and the bugfix has been released in SimPEG v0.22.1. Sorry for not commenting here before. I'm closing this issue now. Thanks again @pjdsmedt for submitting the bug report, much appreciated!
